v22.2/trigram-indexes.md

```diff
@@ -0,0 +1,354 @@
+
+For usage examples, see [Use a trigram index to speed up fuzzy string matching](#use-a-trigram-index-to-speed-up-fuzzy-string-matching).
+
+## Examples
+
+### Simple examples
+
+Suppose you have a table with the following columns:
+
+~~~ sql
+CREATE TABLE t (a INT, w STRING);
+~~~
+
+The following examples illustrate how to create various trigram indexes on column `w`.
+
+A GIN index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~sql
+CREATE INDEX ON t USING GIN (w gin_trgm_ops);
+~~~
+
+A partial index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON t USING GIN (w gin_trgm_ops) WHERE a > 0;
+~~~
+
+A multi-column index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON t USING GIN (a, w gin_trgm_ops);
+~~~
+
+An expression index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON t USING GIN ((lower(w)) gin_trgm_ops);
+~~~
+
+### Use a trigram index to speed up fuzzy string matching
+
+Create a table with a `STRING` column:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE t (w STRING);
+~~~
+
+Populate the table with sample values:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+INSERT INTO t VALUES
+  ('foo'),
+  ('bar'),
+  ('wordy'),
+  ('world'),
+  ('whorl'),
+  ('wort'),
+  ('worm'),
+  ('norm'),
+  ('weird'),
+  ('worried'),
+  ('wofoord'),
+  ('wobarrd');
+
+INSERT INTO t SELECT 'empty' FROM generate_series(1, 10000);
+~~~
+
+Retrieve the columns with values similar to `word`, using the `%` operator. Sort the results by the output of the `similarity()` [built-in function](functions-and-operators.html#trigrams-functions):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT w, similarity(w, 'word')
+  FROM t
+  WHERE w % 'word'
+  ORDER BY similarity DESC, w;
+~~~
+
+~~~
+     w    |     similarity
+----------+----------------------
+  wordy   |  0.5714285714285714
+  wofoord |  0.4444444444444444
+  worm    | 0.42857142857142855
+  wort    | 0.42857142857142855
+  world   |               0.375
+  wobarrd |                 0.3
+  worried |                 0.3
+(7 rows)
+
+
+Time: 30ms total (execution 30ms / network 0ms)
+~~~
+
+Values are not included in the results if their similarities do not meet the threshold set by [`pg_trgm.similarity_threshold`](show-vars.html#pg_trgm_similarity_threshold), which defaults to `0.3`. For example:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT similarity('weird', 'word');
+~~~
+
+~~~
+      similarity
+----------------------
+  0.2222222222222222
+~~~
+
+Notice that the fuzzy search took 30 milliseconds to execute. Without a trigram index, the statement performs a full scan:
```

Comment:
nit: Could be good to refer to the `EXPLAIN` docs page here. "Without a trigram index, the statement performs a full scan. Check this with `EXPLAIN`...: " Or, something along those lines?